Problem with start on Alfresco 25.1

I work with this addon for log time and now I wanna use it in new Alfresco 25.1 inside docker. I get error on start with wrong parse of XML, what is strange. I check format, encoding and change content of XML but non of this work. Is addon tested on Alfresco 25 or is some problem in my installation or build?

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dictionaryModelBootstrap$child#1' defined in URL [jar:file:/usr/local/tomcat/webapps/alfresco/WEB-INF/lib/support-tools-repo-1.2.3.0-SNAPSHOT.jar!/alfresco/module/ootbee-support-tools-repo/module-context.xml]: 03080001 Could not import bootstrap model 'alfresco/module/ootbee-support-tools-repo/model/jsconsole-model.xml'
.....
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dictionaryModelBootstrap$child#1' defined in URL [jar:file:/usr/local/tomcat/webapps/alfresco/WEB-INF/lib/support-tools-repo-1.2.3.0-SNAPSHOT.jar!/alfresco/module/ootbee-support-tools-repo/module-context.xml]: 03080001 Could not import bootstrap model 'alfresco/module/ootbee-support-tools-repo/model/jsconsole-model.xml'
Caused by: org.alfresco.service.cmr.dictionary.DictionaryException: 03080001 Could not import bootstrap model 'alfresco/module/ootbee-support-tools-repo/model/jsconsole-model.xml'
......
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dictionaryModelBootstrap$child#1' defined in URL [jar:file:/usr/local/tomcat/webapps/alfresco/WEB-INF/lib/support-tools-repo-1.2.3.0-SNAPSHOT.jar!/alfresco/module/ootbee-support-tools-repo/module-context.xml]: 03080001 Could not import bootstrap model 'alfresco/module/ootbee-support-tools-repo/model/jsconsole-model.xml'
Caused by: org.alfresco.service.cmr.dictionary.DictionaryException: 03080000 Failed to parse model
......
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dictionaryModelBootstrap$child#1' defined in URL [jar:file:/usr/local/tomcat/webapps/alfresco/WEB-INF/lib/support-tools-repo-1.2.3.0-SNAPSHOT.jar!/alfresco/module/ootbee-support-tools-repo/module-context.xml]: 03080001 Could not import bootstrap model 'alfresco/module/ootbee-support-tools-repo/model/jsconsole-model.xml'
Caused by: org.jibx.runtime.JiBXException: Error parsing document (line 1, col 5)
......
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dictionaryModelBootstrap$child#1' defined in URL [jar:file:/usr/local/tomcat/webapps/alfresco/WEB-INF/lib/support-tools-repo-1.2.3.0-SNAPSHOT.jar!/alfresco/module/ootbee-support-tools-repo/module-context.xml]: 03080001 Could not import bootstrap model 'alfresco/module/ootbee-support-tools-repo/model/jsconsole-model.xml'
Caused by: org.xmlpull.v1.XmlPullParserException: PI must not start with xml (position:unknown ﻿@1:5 in org.jibx.runtime.impl.InputStreamWrapper$WrappedStreamUTF8Reader@233a3aad) 
```

--- a/repository/src/main/java/org/orderofthebee/addons/support/tools/repo/jsconsole/AlfrescoScriptAPITernGet.java
+++ b/repository/src/main/java/org/orderofthebee/addons/support/tools/repo/jsconsole/AlfrescoScriptAPITernGet.java
@@ -77,7 +77,6 @@ import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.DisposableBean;
 import org.springframework.beans.factory.InitializingBean;
 import org.springframework.context.ApplicationContextAware;
-import org.springframework.core.LocalVariableTableParameterNameDiscoverer;
 import org.springframework.core.ParameterNameDiscoverer;
 import org.springframework.extensions.surf.util.I18NUtil;
 import org.springframework.extensions.webscripts.Cache;
@@ -94,6 +93,8 @@ import org.springframework.extensions.webscripts.WebScriptRequest;
 public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements InitializingBean
 {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AlfrescoScriptAPITernGet.class);
+
     private static final Collection<Class<?>> PRIMITIVE_NUMBER_CLASSES = Collections
             .unmodifiableList(Arrays.<Class<?>> asList(byte.class, short.class, int.class, long.class, float.class, double.class));
 
@@ -108,9 +109,42 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
     private static final Collection<String> INIT_METHOD_NAMES = Collections
             .unmodifiableSet(new HashSet<String>(Arrays.<String> asList("init", "register")));
 
-    private static final ParameterNameDiscoverer PARAMETER_NAME_DISCOVERER = new LocalVariableTableParameterNameDiscoverer();
+    private static final ParameterNameDiscoverer PARAMETER_NAME_DISCOVERER;
+    static
+    {
+        // Alfresco 5.0.d (earliest version we want to try to support) contains Spring 3.x
+        // StandardReflectionParameterNameDiscoverer is only available from Spring 4.x on
+        Class<?> cls = null;
+        try
+        {
+            cls = Class.forName("org.springframework.core.StandardReflectionParameterNameDiscoverer");
+        }
+        catch (final ClassNotFoundException ex)
+        {
+            try
+            {
+                cls = Class.forName("org.springframework.core.LocalVariableTableParameterNameDiscoverer");
+            }
+            catch (final ClassNotFoundException ex2)
+            {
+                LOGGER.info("No valid Spring parameter name discoverer class found");
+            }
+        }
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AlfrescoScriptAPITernGet.class);
+        ParameterNameDiscoverer pnd = null;
+        if (cls != null)
+        {
+            try
+            {
+                pnd = (ParameterNameDiscoverer) cls.newInstance();
+            }
+            catch (final InstantiationException | IllegalAccessException ex)
+            {
+                LOGGER.warn("Failed to instantiate Spring paramater name discoverer", ex);
+            }
+        }
+        PARAMETER_NAME_DISCOVERER = pnd;
+    }
 
     protected NamespaceService namespaceService;
 
@@ -320,7 +354,7 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
 
         while (classesToDescribe.size() > classesDescribed.size())
         {
-            final Collection<Class<?>> remainingClasses = new HashSet<Class<?>>(classesToDescribe);
+            final Collection<Class<?>> remainingClasses = new HashSet<>(classesToDescribe);
             remainingClasses.removeAll(classesDescribed);
 
             for (final Class<?> cls : remainingClasses)
@@ -333,7 +367,7 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
 
                     if (skip == null || skip.isEmpty() || !Boolean.parseBoolean(skip))
                     {
-                        final Map<String, Object> typeDefinition = new HashMap<String, Object>();
+                        final Map<String, Object> typeDefinition = new HashMap<>();
                         final Collection<Class<?>> relatedClasses = this.fillClassTypeDefinition(cls, typeDefinition);
                         classesToDescribe.addAll(relatedClasses);
                         typeDefinitions.add(typeDefinition);
@@ -401,7 +435,7 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
             }
         }
 
-        final Collection<Class<?>> dummyClasses = new HashSet<Class<?>>();
+        final Collection<Class<?>> dummyClasses = new HashSet<>();
 
         for (final Entry<String, Object> globalEntry : model.entrySet())
         {
@@ -509,18 +543,18 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
         final String nameOnly = this.properties.getProperty(commonPrefix + ".nameOnly");
         if (nameOnly == null || nameOnly.isEmpty() || !Boolean.parseBoolean(nameOnly))
         {
-            this.fillClassTypeMemberDefinitions(cls, typeDefinition, relatedClasses, commonPrefix, superclass);
+            this.fillClassTypeMemberDefinitions(cls, typeDefinition, relatedClasses, commonPrefix);
         }
 
         return relatedClasses;
     }
 
     protected void fillClassTypeMemberDefinitions(final Class<?> cls, final Map<String, Object> typeDefinition,
-            final Collection<Class<?>> relatedClasses, final String commonPrefix, final Class<?> superclass)
+            final Collection<Class<?>> relatedClasses, final String commonPrefix)
     {
-        final List<Map<String, Object>> memberDefinitions = new ArrayList<Map<String, Object>>();
-        final Map<String, AtomicInteger> usedMemberNames = new HashMap<String, AtomicInteger>();
-        final Collection<String> handledProperties = new HashSet<String>();
+        final List<Map<String, Object>> memberDefinitions = new ArrayList<>();
+        final Map<String, AtomicInteger> usedMemberNames = new HashMap<>();
+        final Collection<String> handledProperties = new HashSet<>();
 
         final List<Method> methods = this.collectDocumentableMethods(cls);
 
@@ -1014,7 +1048,7 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
         typeBuilder.append("fn(");
         if (parameterTypes.length > 0)
         {
-            final String[] parameterNames = PARAMETER_NAME_DISCOVERER.getParameterNames(method);
+            final String[] parameterNames = PARAMETER_NAME_DISCOVERER != null ? PARAMETER_NAME_DISCOVERER.getParameterNames(method) : null;
 
             for (int idx = 0; idx < parameterTypes.length; idx++)
             {
@@ -1047,7 +1081,12 @@ public class AlfrescoScriptAPITernGet extends DeclarativeWebScript implements In
                     typeName = "bool";
                 }
 
-                final String parameterTypeTernName = this.properties.getProperty(methodPrefix + ".arg" + idx + ".typeTernName");
+                String parameterTypeTernName = this.properties.getProperty(methodPrefix + "." + parameterName + ".typeTernName");
+                if ((parameterTypeTernName == null || parameterTypeTernName.isEmpty()) && !parameterName.equals("arg" + idx))
+                {
+                    parameterTypeTernName = this.properties.getProperty(methodPrefix + ".arg" + idx + ".typeTernName");
+                }
+
                 if (parameterTypeTernName != null && !parameterTypeTernName.isEmpty())
                 {
                     typeName = parameterTypeTernName;

--- a/repository/src/main/resources/alfresco/module/ootbee-support-tools-repo/jsconsole-tern.properties
+++ b/repository/src/main/resources/alfresco/module/ootbee-support-tools-repo/jsconsole-tern.properties
@@ -64,9 +64,9 @@ type.org.alfresco.service.ServiceRegistry.nameOnly=true
 type.org.activiti.engine.delegate.VariableScope.nameOnly=true
 
 type.org.alfresco.repo.jscript.ScriptNode.properties.typeTernName=NodeProperties
-type.org.alfresco.repo.jscript.Person.getImmutableProperties.in.String.out.ScriptableHashMap.typeTernName=NodeProperties
+type.org.alfresco.repo.jscript.Person.getImmutableProperties.out.ScriptableHashMap.in.String.typeTernName=NodeProperties
 type.org.alfresco.repo.workflow.jscript.JscriptWorkflowTask.properties.typeTernName=TaskProperties
-type.org.alfresco.repo.jscript.Search.queryResultSet.in.Object.out.Scriptable.typeTernName=SearchResultSetMeta
+type.org.alfresco.repo.jscript.Search.queryResultSet.out.Scriptable.in.Object.typeTernName=SearchResultSetMeta
 
 # native char is handled as a number
 type.java.lang.String.charAt.out.char.in.int.returnTypeTernName=number
@@ -90,38 +90,38 @@ type.org.alfresco.enterprise.repo.management.script.ScriptMBean.setJmxValueConve
 # some methods / properties use a generic Scriptable type for representing native arrays
 type.org.alfresco.repo.jscript.ScriptNode.children.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
 type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.Object.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.Object.int.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.Object.int.int.int.String.Boolean.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.childrenByXPath.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.getChildAssocsByType.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.ScriptNode.getPropertyNames.in.boolean.out.Scriptable.typeClassName=[Ljava.lang.String;
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.Object.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.Object.int.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.Object.int.int.int.String.Boolean.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.childrenByXPath.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.getChildAssocsByType.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.getPropertyNames.out.Scriptable.in.boolean.typeClassName=[Ljava.lang.String;
 type.org.alfresco.repo.jscript.ScriptNode.typePropertyNames.typeClassName=[Ljava.lang.String;
-type.org.alfresco.repo.jscript.ScriptNode.getTypePropertyNames.in.boolean.out.Scriptable.typeClassName=[Ljava.lang.String;
-type.org.alfresco.repo.jscript.Classification.getAllCategoryNodes.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.CategoryNode;
-type.org.alfresco.repo.jscript.Classification.getCategoryUsage.in.String.int.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.Classification$Tag;
-type.org.alfresco.repo.jscript.Classification.getRootCategories.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.CategoryNode;
-type.org.alfresco.repo.jscript.Search.luceneSearch.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.luceneSearch.in.String.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.luceneSearch.in.String.String.boolean.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.luceneSearch.in.String.String.boolean.int.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.luceneSearch.in.String.String.String.boolean.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.luceneSearch.in.String.String.String.boolean.int.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.query.in.Object.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.savedSearch.in.ScriptNode.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.savedSearch.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.selectNodes.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.selectNodes.in.String.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.xpathSearch.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.Search.xpathSearch.in.String.String.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.People.getContainerGroups.in.ScriptNode.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.People.getMembers.in.ScriptNode.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
-type.org.alfresco.repo.jscript.People.getMembers.in.ScriptNode.boolean.out.Scriptable.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.ScriptNode.getTypePropertyNames.out.Scriptable.in.boolean.typeClassName=[Ljava.lang.String;
+type.org.alfresco.repo.jscript.Classification.getAllCategoryNodes.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.CategoryNode;
+type.org.alfresco.repo.jscript.Classification.getCategoryUsage.out.Scriptable.in.String.int.typeClassName=[Lorg.alfresco.repo.jscript.Classification$Tag;
+type.org.alfresco.repo.jscript.Classification.getRootCategories.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.CategoryNode;
+type.org.alfresco.repo.jscript.Search.luceneSearch.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.luceneSearch.out.Scriptable.in.String.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.luceneSearch.out.Scriptable.in.String.String.boolean.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.luceneSearch.out.Scriptable.in.String.String.boolean.int.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.luceneSearch.out.Scriptable.in.String.String.String.boolean.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.luceneSearch.out.Scriptable.in.String.String.String.boolean.int.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.query.out.Scriptable.in.Object.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.savedSearch.out.Scriptable.in.ScriptNode.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.savedSearch.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.selectNodes.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.selectNodes.out.Scriptable.in.String.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.xpathSearch.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.Search.xpathSearch.out.Scriptable.in.String.String.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.People.getContainerGroups.out.Scriptable.in.ScriptNode.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.People.getMembers.out.Scriptable.in.ScriptNode.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
+type.org.alfresco.repo.jscript.People.getMembers.out.Scriptable.in.ScriptNode.boolean.typeClassName=[Lorg.alfresco.repo.jscript.ScriptNode;
 # don't know why, but getPeople returns array of NodeRef instead of ScriptNode
-type.org.alfresco.repo.jscript.People.getPeople.in.String.out.Scriptable.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
-type.org.alfresco.repo.jscript.People.getPeople.in.String.int.out.Scriptable.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
-type.org.alfresco.repo.jscript.People.getPeople.in.String.int.String.boolean.out.Scriptable.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
+type.org.alfresco.repo.jscript.People.getPeople.out.Scriptable.in.String.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
+type.org.alfresco.repo.jscript.People.getPeople.out.Scriptable.in.String.int.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
+type.org.alfresco.repo.jscript.People.getPeople.out.Scriptable.in.String.int.String.boolean.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
 type.org.alfresco.repo.jscript.People.getPeoplePaging.in.String.ScriptPagingDetails.String.Boolean.out.Scriptable.typeClassName=[Lorg.alfresco.service.cmr.repository.NodeRef;
 type.org.alfresco.repo.workflow.jscript.WorkflowManager.assignedTasks.typeClassName=[Lorg.alfresco.repo.workflow.jscript.JscriptWorkflowTask;
 type.org.alfresco.repo.workflow.jscript.WorkflowManager.completedTasks.typeClassName=[Lorg.alfresco.repo.workflow.jscript.JscriptWorkflowTask;
@@ -134,14 +134,23 @@ type.org.alfresco.repo.workflow.jscript.JscriptWorkflowPath.tasks.typeClassName=
 
 # some methods expect native-like objects
 # childFileFolders supports JavaString, native String and native array of native String - we opt for the latter for documentation
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.Object.out.Scriptable.arg2.typeTernName=[string]
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.Object.int.out.Scriptable.arg2.typeTernName=[string]
-type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.in.boolean.boolean.Object.int.int.int.String.Boolean.String.out.Scriptable.arg2.typeTernName=[string]
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.Object.arg2.typeTernName=[string]
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.Object.int.arg2.typeTernName=[string]
+type.org.alfresco.repo.jscript.ScriptNode.childFileFolders.out.Scriptable.in.boolean.boolean.Object.int.int.int.String.Boolean.String.arg2.typeTernName=[string]
+# createNode supports a native Object-like structure for properties
+type.org.alfresco.repo.jscript.ScriptNode.createNode.out.ScriptNode.in.String.String.Object.arg2.typeTernName=object
+type.org.alfresco.repo.jscript.ScriptNode.createNode.out.ScriptNode.in.String.String.Object.String.arg2.typeTernName=object
+type.org.alfresco.repo.jscript.ScriptNode.createNode.out.ScriptNode.in.String.String.Object.String.String.String.arg2.typeTernName=object
+# addAspect supports a native Object-like structure for properties
+type.org.alfresco.repo.jscript.ScriptNode.addAspect.out.boolean.in.String.Object.arg1.typeTernName=object
+# processTemplate supports a native Object-like structure for arguments
+type.org.alfresco.repo.jscript.ScriptNode.processTemplate.out.String.in.ScriptNode.Object.arg1.typeTernName=object
+type.org.alfresco.repo.jscript.ScriptNode.processTemplate.out.String.in.String.Object.arg1.typeTernName=object
 
 type.org.springframework.extensions.webscripts.ConfigModel.script.typeClassName=java.lang.String
 
-type.org.springframework.extensions.webscripts.ScriptMessage.get.in.String.Scriptable.out.String.arg1.typeClassName=[Ljava.lang.String;
-type.org.springframework.extensions.webscripts.ScriptMessage.get.in.String.Scriptable.out.String.arg1.typeTernName=[string]
+type.org.springframework.extensions.webscripts.ScriptMessage.get.out.String.in.String.Scriptable.arg1.typeClassName=[Ljava.lang.String;
+type.org.springframework.extensions.webscripts.ScriptMessage.get.out.String.in.String.Scriptable.arg1.typeTernName=[string]
 
 # default doc without I18n
 type.java.lang.String.ternDoc=Java representation of a string value not wrapped into a native Rhino / JS string

